internal/handler/authorize.go

```diff
@@ -1,52 +1,33 @@
 package handler
 
 import (
-	"sort"
-
 	"github.com/dhis2-sre/im-manager/pkg/model"
 	"github.com/dhis2-sre/im-user/swagger/sdk/models"
+	"golang.org/x/exp/slices"
 )
 
 const AdministratorGroupName = "administrators"
 
 func CanReadInstance(user *models.User, instance *model.Instance) bool {
-	return isAdministrator(user) || isMemberOf(instance.GroupID, user.Groups)
+	return isAdministrator(user) || isMemberOf(instance.GroupName, user.Groups)
 }
 
 func CanWriteInstance(user *models.User, instance *model.Instance) bool {
 	return isAdministrator(user) ||
-		isMemberOf(instance.GroupID, user.AdminGroups) ||
-		(isOwner(user, instance) && isMemberOf(instance.GroupID, user.Groups))
+		isMemberOf(instance.GroupName, user.AdminGroups) ||
+		(isOwner(user, instance) && isMemberOf(instance.GroupName, user.Groups))
 }
 
 func isOwner(user *models.User, instance *model.Instance) bool {
 	return uint(user.ID) == instance.UserID
 }
 
-func isMemberOf(groupId uint, groups []*models.Group) bool {
-	sort.Slice(groups, func(i, j int) bool {
-		return groups[i].ID <= groups[j].ID
-	})
-
-	index := sort.Search(len(groups), func(i int) bool {
-		return uint(groups[i].ID) >= groupId
-	})
-
-	return index < len(groups) && uint(groups[index].ID) == groupId
+func isMemberOf(groupName string, groups []*models.Group) bool {
+	f := func(g *models.Group) bool { return g.Name == groupName }
```

Comment:
Same as in im-user (I think that's where we have the same). Inline or replace with range without the need for dependency (I would prefer it that way).